### ISSUE TITLE
`poetry install` checks `README.md`, but `poetry check` does not.

### ISSUE THREAD

**Issue by YDX-2147483647** [python-poetry/poetry#6937]
- **Poetry version**: 1.2.2
- **Python version**:  3.10.8
- **OS version and name**: Windows 10 (21H1) and Raspberry Pi OS (based on Debian 11 Bullseye)
- **pyproject.toml**: default – `poetry new <package_name>` can fire this issue.

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes. (*Note*: I’ve only checked FAQ and blog 1.2.x)
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

<u>`poetry install` checks `README.md`, but `poetry check` does not.</u>

>  To be rigorous, it checks [`tool.poetry.readme`](https://python-poetry.org/docs/pyproject/#readme) in `pyproject.toml`.

## Is it an issue?

- **All commands work**: `poetry install` raises a `FileNotFoundError`, but it works.

- [Doc](https://python-poetry.org/docs/cli/#install): The `install` command reads `pyproject.toml`, resolves the dependencies, and installs them.

  It should not care about `README`.

- [Doc](https://python-poetry.org/docs/cli/#check): The `check` command validates the structure of  `pyproject.toml`.

  Although `README` is absent, the TOML structure is valid. So this might be expected.

## To Reproduce

TL; DR:

1. Create a project.
2. `rm README.md`.
3. `poetry install`: No such file or directory: `README.md`.

### 1. Create a project

```shell
$ poetry new why-read-me
$ cd why-read-me

$ cat pyproject.toml
# There's a default `README.md`.
[tool.poetry]
name = "why-read-me"
……
readme = "README.md"
……
```

### 2. `poetry install` checks `README.md`

```shell
$ mv .\README.md read-me-whatever.md
# Let's steal the README.

$ poetry install
# The `install` command checks the README, and raises an error.
# However, it successfully installs packages. (If I add some packages in pyproject.toml, it will be installed here.)
Using virtualenv: ……
Installing dependencies from lock file

Finding the necessary packages for the current system

  FileNotFoundError

  [Errno 2] No such file or directory: '……/why-read-me/README.md'

  at ……\Python310\lib\pathlib.py:1119 in open
      1115│         the built-in open() function does.
      1116│         """
      1117│         if "b" not in mode:
      1118│             encoding = io.text_encoding(encoding)
    → 1119│         return self._accessor.open(self, mode, buffering, encoding, errors,
      1120│                                    newline)
      1121│
```

### 3. Other commands work

```shell
$ poetry check
# It ignores the absence of README.
All set!

$ poetry add any-package
Using version …… for ……
Updating dependencies
Writing lock file
Package operations: …… installs, ……
  - Installing …… (……)
……
# Nothing broken and No error raised.
```

### 4. Update`tool.poetry.readme`

```shell
$ nano pyproject.toml
# Change `tool.poetry.readme` to `read-me-whatever.md`.

$ poetry install
# Everthing is OK now.
Installing dependencies from lock file
Installing the current project: why-read-me (0.1.0)
```

## Full output with `-vv`

<details>
<summary>Output of `poetry install -vv`</summary>
Using virtualenv: ……\temp\why-read-me\.venv
Installing dependencies from lock file

Finding the necessary packages for the current system

  Stack trace:

  11  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\cleo\application.py:329 in run
        exit_code = self._run(io)

  10  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\poetry\console\application.py:185 in _run
        exit_code: int = super()._run(io)

   9  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\cleo\application.py:423 in _run
        exit_code = self._run_command(command, io)

   8  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\cleo\application.py:465 in _run_command
        raise error

   7  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\cleo\application.py:449 in _run_command
        exit_code = command.run(io)

   6  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\cleo\commands\base_command.py:119 in run
        status_code = self.execute(io)

   5  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\cleo\commands\command.py:83 in execute
        return self.handle()

   4  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\poetry\console\commands\install.py:155 in handle
        builder = EditableBuilder(self.poetry, self.env, self.io)

   3  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\poetry\masonry\builders\editable.py:47 in __init__
        super().__init__(poetry)

   2  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\poetry\core\masonry\builders\builder.py:90 in __init__
        self._meta = Metadata.from_package(self._package)

   1  ~\AppData\Roaming\pypoetry\venv\lib\site-packages\poetry\core\masonry\metadata.py:60 in from_package
        with readme.open(encoding="utf-8") as f:

  FileNotFoundError

  [Errno 2] No such file or directory: '……\\temp\\why-read-me\\README.md'

  at C:\Program Files\Python310\lib\pathlib.py:1119 in open
      1115│         the built-in open() function does.
      1116│         """
      1117│         if "b" not in mode:
      1118│             encoding = io.text_encoding(encoding)
    → 1119│         return self._accessor.open(self, mode, buffering, encoding, errors,
      1120│                                    newline)
      1121│
      1122│     def read_bytes(self):
      1123│         """
</details>

**Comment by Amarettini** [comment]
I thought there was a general problem with my project structure as a newbie to poetry. I wonder how such problem has gone unnoticed till now. Spent way too much time figuring out the problem source.

**Comment by YDX-2147483647** [comment]
`poetry install` need metadata, whose `description` is from readmes. (According to “Full output with `-vv`”)

```python
# poetry/core/masonry/metadata.py
class Metadata:
    def from_package(cls, package: Package) -> Metadata:
        meta = cls()

        if package.readmes:
            descriptions = []
            for readme in package.readmes:
                with readme.open(encoding="utf-8") as f:  # ⇒ FileNotFoundError
                    descriptions.append(f.read())
            meta.description = "\n".join(descriptions)
```

https://github.com/python-poetry/poetry-core/blob/978fb55a2f78e1e11cd226722713c70a102dc40d/src/poetry/core/masonry/metadata.py#L48-L62

**Comment by YDX-2147483647** [comment]
As discussed in #7424, `poetry check` should check everything.

```python
if "readme" in config:
    readmes = config["readme"]
    if isinstance(config["readme"], str):
        readmes = (config["readme"],)

    # Checking types of all readme files (must match)
    readme_types = {readme_content_type(r) for r in readmes}
    if len(readme_types) > 1:
        # ……

    # Checking existence of readme files
    missing_readmes = [r for r in readmes if not Path(r).exists()]
    if missing_readmes:
        result["errors"].append(
            "Cannot find declared README file(s):"
            f" {', '.join(missing_readmes)}"
        )
```

I've tried to modify `poetry/core/factory.py` (as above), but I find it a bad idea… `Factory` is used as an _abstract_ factory in `test_factory.py`, so the following tests will be broken.

- `test_validate_strict_fails_strict_and_non_strict`
- `test_strict_validation_success_on_multiple_readme_files`
- `test_strict_validation_fails_on_readme_files_with_unmatching_types`
